### PR TITLE
Impl items have generics

### PR DESCRIPTION
**Pull request by oli-obk**

### CODE REVIEW

File: src/librustc_mir/const_eval.rs [hunk 1]
```diff
@@ -525,7 +525,8 @@ pub fn const_eval_provider<'a, 'tcx>(
     tcx: TyCtxt<'a, 'tcx, 'tcx>,
     key: ty::ParamEnvAnd<'tcx, GlobalId<'tcx>>,
 ) -> ::rustc::mir::interpret::ConstEvalResult<'tcx> {
-    trace!("const eval: {:?}", key);
+    trace!("const eval: {}", key.value.instance);
+    trace!("const eval key: {:?}", key);
```
Comment (bjorn3) [inline src/librustc_mir/const_eval.rs]: Why trace both `key` and `key.value.instance`?

**Review by bjorn3 — CHANGES_REQUESTED** [review]

File: src/librustc_mir/const_eval.rs (hunk 1 above)
Comment (oli-obk) [reply src/librustc_mir/const_eval.rs]: Because the printing via `Display` of the instance goes through the code that was ICEing here before

Comment (bjorn3) [reply src/librustc_mir/const_eval.rs]: Get it

**Comment by RalfJung** [comment]
You just rebased, right? So we can r=davidtwco again?

Cc @davidtwco

**Comment by davidtwco** [comment]
Thanks @RalfJung. Wish GitHub would email on pushes.

@bors r+

**Comment by oli-obk** [comment]
rebased again, and added a comment about what's going on

**Pull request merged**